tuning/sft_trainer.py

```diff
@@ -461,12 +461,46 @@ def parse_arguments(parser, json_config=None):
     )
 
 
-def main(**kwargs):  # pylint: disable=unused-argument
-    logger = logging.get_logger("__main__")
+def set_log_level(parsed_training_args):
+    """Set log level of python native logger and TF logger via argument from CLI or env variable.
+
+    Args:
+        parsed_training_args
+            Training arguments for training model.
+    """
+
+    # Clear any existing handlers if necessary
+    for handler in logging.root.handlers[:]:
+        logging.root.removeHandler(handler)
+
+    # Configure Python native logger log level
+    # If CLI arg is passed, assign same log level to python native logger
+    if parsed_training_args.log_level != "passive":
+        logging.basicConfig(level=parsed_training_args.log_level.upper())
+    else:
+        # Assign value of either env var LOG_LEVEL or warning
+        LOGLEVEL = os.environ.get("LOG_LEVEL", "WARNING").upper()
+        logging.basicConfig(level=LOGLEVEL)
+        train_logger = logging.getLogger()
```

Comment:
Could this work to attain the logic we're looking for?
Goal for logic:
- TRANSFORMERS_VERBOSITY ---> should only set transformers logging, this is different than setting --log_level
- `--log_level` --> sets both python and transformers logging
- For consistency with the flag, LOG_LEVEL also sets python and transformers logging

```py
    if parsed_training_args.log_level != "passive":
        logging.basicConfig(level=parsed_training_args.log_level.upper())
    elif os.environ.get("LOG_LEVEL"):
        # Assign value of either env var LOG_LEVEL or warning
        LOGLEVEL = os.environ.get("LOG_LEVEL")
        logging.basicConfig(level=LOGLEVEL)
       # Set log_level in TrainingArguments
       parsed_training_args.log_level = LOGLEVEL.lower()

  train_logger = logging.getLogger()
```

Reply:
I didn't add this line `parsed_training_args.log_level = LOGLEVEL.lower()` in the same if-else because if env variable 'TRANSFORMERS_VERBOSITY' is set, then this would get overwritten. Hence wrote this line in separate log below checking `if not os.environ.get("TRANSFORMERS_VERBOSITY"):`

But yea the code that you write here is optimized. Thank you!

Reply:
And yes it does achieve the logic we aim for. Tested locally and on image too!

Reply:
Let's chat about this logic to clarify